### PR TITLE
feat: fix order and focused item

### PR DESCRIPTION
**Pull request by morgilad**
Match the IntelliSense behavior as in KE. 
This includes:

- Maintaining the sort order of completion items even when there is user input.
- Focusing on the most relevant item based on the user input.

### CODE REVIEW

File: package/tests/integration/completion-items.spec.ts [hunk 1]
```diff
@@ -40,23 +40,47 @@ test.describe('completion items', () => {
...
+        await model.intellisense().wait();
+        await expect(model.intellisense().option(0).locator).toContainText('count()');
+        await expect(model.intellisense().option(1).locator).toContainText('count_distinct(');
+    });
+
+    test('keeps language-service ordering', async ({ page }) => {
+        await page.keyboard.type('where t');
+        await model.intellisense().wait();
+
+        await page.keyboard.type('ime');
```
Comment (morgilad) [inline package/tests/integration/completion-items.spec.ts:55]: explain why the split is here

File: README.md [hunk 2]
```diff
@@ -85,6 +85,10 @@ Every PR should come with a test that checks it.
 
 ## Changelog
 
+### 11.4.0
+
+-   feat: IntelliSense completion items now maintain the original order from the language service, with the most relevant option automatically focused based on the user's input.
```
Comment (morgilad) [inline README.md:90]: add that we keep KE order

File: package/src/languageFeatures.utils.ts [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
-    const containedInFilterText = completionItem.filterText.toLowerCase().includes(searchWord.toLowerCase());
-    const shouldPrependSearchWord = completionItem.data.forcePrecedence && containedInFilterText;
+    const containedInFilterText = completionItem.filterText.toLowerCase().includes(userInput.toLowerCase());
+    if (!containedInFilterText) return completionItem.filterText;
 
-    return shouldPrependSearchWord ? `${searchWord}${completionItem.filterText}` : completionItem.filterText;
+    return `${userInput}${completionItem.filterText}`;
+}
+
+export function getSelectedIndex(completionItems: ls.CompletionItem[], userInput: string): number {
```
Comment (morgilad) [inline package/src/languageFeatures.utils.ts:12]: make sure the first item isnt selected when no item should be selected

File: package/src/languageFeatures.ts [hunk 4]
```diff
@@ -814,27 +814,30 @@ export class CompletionAdapter implements monaco.languages.CompletionItemProvide
...
 
         return this.completionCacheManager
-            .getCompletionItems(word, resource, fromPosition(position))
+            .getCompletionItems(userInput, resource, fromPosition(position))
             .then((info) => (onDidProvideCompletionItems ? onDidProvideCompletionItems(info) : info))
             .then((info) => {
                 if (!info) return;
 
-                let items: monaco.languages.CompletionItem[] = info.items.map((entry) => {
+                const selectedItemIndex = getSelectedIndex(info.items, userInput);
```
Comment (morgilad) [inline package/src/languageFeatures.ts:827]: replace index with item

**Review by sagivf — APPROVED** [review]
LGTM 🎈

**Pull request merged**